pkg/controller/direct/spanner/instance_controller.go

```diff
@@ -151,6 +152,14 @@ func (a *SpannerInstanceAdapter) Create(ctx context.Context, createOp *directbas
 		return err
 	}
 	resource := SpannerInstanceSpec_ToProto(mapCtx, &desired.Spec, a.id.SpannerInstanceConfigPrefix())
+
+	// if edition is unspecified, Spanner API default edition to STANDARD.
+	// Spanner API bans downgrade edition from STANDARD to unspecifed,
+	// which cause infinite reconcile loop.
+	if desired.Spec.Edition == nil {
+		resource.Edition = spannerpb.Instance_STANDARD
+	}
+
```

Comment:
Suggest leaving the GCP `edition` as unset, hopefully  the spanner server will use the STANDARD value, but it looks like there's a chance it could be another value.